File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/Rect.kt
package dev.hebei.camerax_android.legacy

class Rect(registrar: CameraXRegistrar) : PigeonApiRect(registrar) {
    override fun pigeon_defaultConstructor(left: Long, top: Long, right: Long, bottom: Long): android.graphics.Rect {
        return android.graphics.Rect(left.toInt(), top.toInt(), right.toInt(), bottom.toInt())
    }

    override fun getWidth(pigeon_instance: android.graphics.Rect): Long {
        return pigeon_instance.width().toLong()
    }

    override fun getHeight(pigeon_instance: android.graphics.Rect): Long {
        return pigeon_instance.height().toLong()
    }
}

File: camerax_android/lib/src/legacy/common/rect.dart
import 'package:camerax_android/src/legacy/camerax.g.dart' as $native;
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;

extension RectObj on $native.Rect {
  $interface.Rect get args {
    return $interface.Rect(left, top, right, bottom);
  }
}
